src/libYARP_OS/include/yarp/os/Carrier.h

```diff
@@ -15,6 +15,8 @@
 #include <yarp/os/ConnectionReader.h>
 #include <yarp/os/Connection.h>
 #include <yarp/os/ConnectionState.h>
+#include <yarp/os/Face.h>
+#include <yarp/os/impl/TcpFace.h> // to move in cpp file
```

Comment:
Sorry, I just noticed this after @traversaro comment, it is not ok to have "impl" headers included in "not-impl" headers, please move this in the .cpp file

Reply:
All functions have been implemented in header file. So I'll move all implementations to .cpp file.

Reply:
Done. see 7d355df16ee4f1f68a34482e96175836f1dceec9